lib/wovnrb/html_replacers/meta_replacer.rb

```diff
@@ -1,15 +1,22 @@
 module Wovnrb
   class MetaReplacer < ReplacerBase
-    def initialize(text_index)
+    def initialize(text_index, pattern = nil, headers = nil)
       @text_index = text_index
+      @pattern = pattern
+      @headers = headers
     end
 
     def replace(dom, lang)
       dom.xpath('//meta').select { |node|
         next if wovn_ignore?(node)
-        (node.get_attribute('name') || node.get_attribute('property') || '') =~ /^(description|title|og:title|og:description|twitter:title|twitter:description)$/
+        (node.get_attribute('name') || node.get_attribute('property') || '') =~ /^(description|title|og:title|og:description|og:url|twitter:title|twitter:description)$/
       }.each do |node|
         node_content = node.get_attribute('content').strip
+        if node.get_attribute('property') && node.get_attribute('property') === 'og:url' && @headers && @pattern
```

Comment:
`if @headers && @pattern && node.get_attribute('property') === 'og:url'` might be better imo, because @headers is default nil -> we skip this check always for other replacer than MetaReplacer